src/main/java/net/minecraftforge/common/util/ITeleporter.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2017.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.util;
+
+import net.minecraft.entity.Entity;
+
+/**
+ * Interface for handling the placement of entities during dimension change.
+ */
+public interface ITeleporter
+{
+    /**
+     * Called to handle placing the entity in the new world.
+     * (e.g. finding a safe place, adding a way back, etc)
```

Comment:
can we have some clarification regarding the preconditions of this method?

e.g. at calling time...

* Which world is the entity in?
* What position? At its old world coords, at the coords after calculating the movement factor, at somewhere else?

Reply:
The position is at the coords with the movement factor applied.

The entity is somewhat in 'limbo' at this point.

If it's desired, I can update the interface to take an explicit world reference, which might simplify things a bit.

Reply:
I was mainly asking for a comment on the interface method stating the answers to those questions, etc.

Reply:
Yes, things are not very clear to a newcomer to this interface like me